crypto/x509/algorithm.c

```diff
@@ -97,8 +97,8 @@ int x509_digest_sign_algorithm(EVP_MD_CTX *ctx, X509_ALGOR *algor) {
   }
 
 #ifdef ENABLE_DILITHIUM
-  if (EVP_PKEY_id(pkey) == EVP_PKEY_DILITHIUM3) {
-    return X509_ALGOR_set0(algor, OBJ_nid2obj(NID_DILITHIUM3_R3), V_ASN1_UNDEF, NULL);
+  if (EVP_PKEY_id(pkey) == EVP_PKEY_PQDSA) {
+    return X509_ALGOR_set0(algor, OBJ_nid2obj(NID_MLDSA65), V_ASN1_UNDEF, NULL);
```

Comment:
how is this supposed to work when you add other PQDSA algorithms?

Reply:
Thank you, found an elegant solution in [369080b](https://github.com/aws/aws-lc/pull/1963/commits/369080b5a595f1880b4edc75a12eb8dcbe643c02). The `pqdsa` struct will hold information on the specific PKEY type. We could have used `EVP_PKEY_base_id` for this, if we didn't disable it in AWS-LC. Nevertheless, if the `PKEY` type is `EVP_PKEY_PQDSA` as in this conditional, we can get the actual NID from `pkey->pkey.pqdsa_key->pqdsa->nid`.